### PR TITLE
allow linting of individual files (#584)

### PR DESCRIPTION
**Pull request by chathaway-codes**
Add lint option to specify linting of specific file. Optional flag `-l` followed by a file-path indicates file to lint.

## Additions

-

## Removals

-

## Changes

-

## Testing

1. Bench testing; ran by hand

## Notes

I'm not sure how straightforward it would be to add an additional task for a non-qualified 'file' argument in gulp; this seems to work, and wasn't too much effort.

I'm not a JS developer; if I got it wrong, please comment.

## Todos

-

## Checklist

- [X] PR has an informative and human-readable title
- [X] Changes are limited to a single goal (no scope creep)
- [X] Code can be automatically merged (no conflicts)
- [X] Code follows the standards laid out in the [CFPB development guidelines](https://github.com/cfpb/development)
- [ ] Passes all existing automated tests
- [X] Any _change_ in functionality is tested
- [X] New functions are documented (with a description, list of inputs, and expected output)
- [X] Placeholder code is flagged / future todos are captured in comments
- [N/A] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [N/A] Project documentation has been updated
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Browsers

- [ ] Chrome on desktop
- [ ] Firefox
- [ ] Safari on macOS
- [ ] Edge
- [ ] Internet Explorer 9, 10, and 11
- [ ] Safari on iOS
- [ ] Chrome on Android

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

File: gulp/tasks/lint.js [hunk 1]
```diff
@@ -47,9 +47,17 @@ gulp.task( 'lint:scripts', () => _genericLint( configLint.src ) );
...
-  gulp.parallel(
-    'lint:build',
-    'lint:tests',
-    'lint:scripts'
-  )
+  // Checks to see if a flag for a specific file is present, if so, use it
+  // Else, returns a list of sane default tasks to run
+  (() => {
+    let argv = minimist( process.argv.slice( 3 ) );
+    if(argv.l !== undefined) {
```
Comment (anselmbradford) [inline gulp/tasks/lint.js]: You should be able to fix the TravisCI failure by changing this to:

```
let UNDEFINED;
if ( argv.l !== UNDEFINED ) {
```

For better or worse, we have https://eslint.org/docs/rules/no-undefined set.

**Comment by anselmbradford** [comment]
Thanks very much for the contribution @chathaway-codes. Two thoughts:

- I presume you chose `l` as the argument for the individual file for the "l" in "lint"? I wonder how you feel about something a little more direct, especially since "l" might look like "I" (eye). Perhaps spell it out "file", like is done with the "fix" flag. Or could the path be passed in directly without a flag? So if there was a path it would lint only that file, if there wasn't it would lint everything?

- We'd want to add documentation for individual linting to https://github.com/cfpb/cfgov-refresh/blob/2f256ab653aaa2eaa3e6443c0471cf8c8ef24abe/docs/browser-acceptance-tests.md#source-code-linting

**Comment by anselmbradford** [comment]
Cross-referencing associated issue https://github.com/cfpb/cfgov-refresh/issues/584

**Comment by chathaway-codes** [comment]
`-l` was select because `-f` was taken by Gulp (and `--file`, I presume). I had trouble getting it to be passed without a flag (that was my first choice also), and working on this was just a late-night one-off. 

How about `--lint``?

And sure thing, I can add documentation to that file.

**Comment by anselmbradford** [comment]
> -l was select because -f was taken by Gulp (and --file, I presume). I had trouble getting it to be passed without a flag (that was my first choice also), and working on this was just a late-night one-off.
>
> How about --lint?

I appreciate it! Is `--path` available? Otherwise `--lint` is fine.

**Comment by chathaway-codes** [comment]
For the record, `--path` was available, and I did change it to use that. Please let me know if anything else is needed!

Good luck!

**Comment by anselmbradford** [comment]
Works for me. I tested e.g. `gulp lint --path=cfgov/unprocessed/apps/owning-a-home/js/form-explainer/form-explainer.js`. I'll double-check something and then merge.

**Pull request merged**